src/io.rs

```diff
@@ -274,11 +301,28 @@ pub trait Write {
     fn remaining_space(&self) -> usize;
 
     /// Writes a little-endian integer.
-    fn write_le<I: LeInt>(&mut self, val: I) -> Result<(), Error> {
+    fn write_le<I: LeInt>(&mut self, val: I) -> Result<(), Error>
+    where
+        Self: Sized,
+    {
         val.write_to(self)
     }
 }
 
+assert_obj_safe!(Write);
+
+impl<W: Write + ?Sized> Write for &'_ mut W {
+    #[inline]
+    fn write_bytes(&mut self, buf: &[u8]) -> Result<(), Error> {
+        W::write_bytes(*self, buf)
+    }
+
+    #[inline]
+    fn remaining_space(&self) -> usize {
+        W::remaining_space(*self)
+    }
```

Comment:
Again, don't you want to pass `write_le` down to `W::write_le`?

Reply:
See above.